rusk-wallet/src/clients.rs

```diff
@@ -119,9 +119,15 @@ impl State {
     }
 
     pub(crate) fn cache(&self) -> Arc<Cache> {
-        let state = self.cache.lock().unwrap();
-
-        Arc::clone(&state)
+        let state = self.cache.lock();
+
+        // We can get an error if the thread holding the lock panicked while
+        // holding the lock. In this case, we can recover the guard from the
+        // poison error and return the guard to the caller.
+        match state {
+            Ok(guard) => Arc::clone(&guard),
+            Err(poisoned) => Arc::clone(&poisoned.into_inner()),
```

Comment:
This is a crate fn, if its called more than once we'll just deadlock anyways 

The lock is also dropped once the function returns so im not sure if we need the match here. The function is also synchronous it won't be in a multithreaded environment

Reply:
We can get the error if the previous user of the mutex panicked while holding it, no matter was the user from the current thread or another one. We need to handle this possible error either by resetting the cache, or by recovering it. It is safer to to leave the match here.